sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,337 @@
+#  Copyright (c) Akretion 2020
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from copy import deepcopy
+
+from marshmallow_objects import ValidationError as MarshmallowValidationError
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.component.core import Component
+
+MAPPINGS_SALE_ORDER_ADDRESS_SIMPLE = [
+    "name",
+    "street",
+    "street2",
+    "zip",
+    "city",
+    "email",
+]
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "processor"
+    _name = "importer.sale.channel"
+    _apply_on = ["sale.order"]
+    _usage = "json_import"
+
+    def run(self, raw_data):
+        """
+        :param raw_data: json-like string
+        :return: generated sale order
+        """
+        # we need this step because the chunk data is stored as
+        # a string (so we can edit it)
+        try:
+            so_datamodel_load = self.env.datamodels["sale.order"].load_json(raw_data)
+        except MarshmallowValidationError:
+            raise ValidationError(MarshmallowValidationError)
+        except Exception:
+            raise
+        json_data_initial = so_datamodel_load.dump()
+        json_data = deepcopy(json_data_initial)
+        so_vals = self.process_data(json_data)
+        new_sale_order = self.env["sale.order"].create(so_vals)
+        self.finalize(new_sale_order, json_data_initial)
+        return new_sale_order
+
+    # DATAMODEL PROCESSORS
+    def process_data(self, so_vals):
+        """ Transform values in-place
+         to make it usable in create() """
+        self.process_simple_fields(so_vals)
```

Comment:
maybe it will be better and simplier to have let method and a _prepare_vals method that look like something like

```
def _prepare_vals(self, data):
    partner = self._process_partner(data)
    invoice_address = self._process_address(partner, data["address_invoicing"])
    invoice_address = self._process_address(partner, data["address_shipping"])
   
    return {
        "partner_id": partner.id,
        "partner_invoice_id": partner_invoice.id,
        "partner_shipping_id": partner_shipping.id,
        "si_amount_total": data["amout_total"],
        "si_amount_tota_untaxed": data["amout_total_untaxed"],
        "order_line": [(0, 0, self._prepare_sale_line(line) for line in data["lines"]]
}
``